### PR TITLE
New design: adds program page

### PR DESCRIPTION
**Pull request by jkachel**
# What are the relevant tickets?

Part of #1773 
https://github.com/mitodl/hq/issues/1947
https://github.com/mitodl/hq/issues/1544

# Description (What does it do?)

Adds the program page layout to the system, including the course selector. 

# Screenshots (if appropriate):

program-page-full

Enroll Now before course selection:

Screenshot 2023-09-07 at 2 52 02 PM

Enroll Now after selection:

Screenshot 2023-09-07 at 2 52 07 PM

# How can this be tested?

For this to work, you will need the `mitxonline-new-product-page` flag enabled, and ideally a program page that has been filled out. Note that program pages will be mostly empty unless you've put info in them - since we had the pages but weren't exposing them, there wasn't a need to have these ready with data.

Testing is simply visiting a program page. Program pages are linked to from course pages. All operations should work as expected and the desktop design should match the Figma layout.

Additionally:
- The course requirement breakdowns use the title of their respective nodes from the requirements tree. If they are empty strings, they will default to "Electives" and "Core Courses". 
- The price is always free. 
- The certificate track pricing information comes from the Price block on the Program page - we (still) don't allow you to purchase an upgrade for an entire program, so this is left for the admin to specify. 
- The enrollment screen hides the certificate track information until a course is selected (as there's no data for that otherwise). Selecting a course will pull in the certificate track information and financial assistance information for that course. 
- This does not automatically enroll you in the program since you'll get an enrollment when you choose and enroll in the course.

# Additional Context

There's a couple edge cases that I don't think were planned for:
- If all courses are audit-only, we still ask them to upgrade. This will also probably break the selector. I wasn't sure what to do here. (This is probably fine to leave alone, though, since I don't know that this is a valid state anyway.)
- Similarly, there's not a good option if the learner selects a program that doesn't have an upgrade path (i.e. most have upgrades, but one doesn't for whatever reason). 
- The start date is just the earliest start date of the courses that are there. 

Additionally:
- This fixes a regression in the sidebar CSS - the max-width of that part of the app should be 400px, not 350, and that's fixed in this PR. 
- The price display in the Course Enrollment is now fixed in this PR (it was not considering approved flexible pricing).
- This removes `font-face` directives in its CSS - it should use the defaults. (On my system, this means it did not pick up the new fonts we're using.)

### CODE REVIEW

**Comment by annagav** [comment]
@jkachel I am going to unassaign myself because my program setup is completely broken, and I don't want to hold up the review.

**Review by rachellougee — COMMENTED** [review]
I left some small comments, otherwise it look good. I will do a functional test.

**Comment by rachellougee** [comment]
@jkachel Can you fix the merge conflict and do a rebase? For some reason, the video is really big on this branch, it seems like it's missing some css in product-details.scss.

image

I am testing the functionality for program page

File: courses/api.py [hunk 1]
```diff
@@ -117,6 +118,54 @@ def get_user_relevant_course_run_qset(
...
+        run_qset = run_qset.annotate(verified_enrollments=verified_enrollments)
+
+        runs = run_qset.filter(
+            Q(user_enrollments__gt=0)
+            | Q(enrollment_end=None)
+            | Q(enrollment_end__gt=now)
+        )
+    else:
+        runs = run_qset.filter(
+            Q(enrollment_end=None) | Q(enrollment_end__gt=now)
```
[the remark this answers is not in the document]
Comment (jkachel) [reply courses/api.py]: This filter gets expanded to include the learner's enrollments if there's an active session.

File: cms/templatetags/course_list.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+@register.inclusion_tag("course_list_card.html", name="course_list")
+def course_list(courses):
+    cards = []
+
+    for course in courses:
+        start_descriptor = (
+            f"Starts {format_course_start_time(course.first_unexpired_run.start_date)}"
+            if course.first_unexpired_run.start_date
```
Comment (rachellougee) [inline cms/templatetags/course_list.py:36]: I am getting 500 error `AttributeError: 'NoneType' object has no attribute 'start_date'` on this line for programs where one of courses have no open run. The program is linked from homepage and catalog page

File: courses/views/v1/__init__.py [hunk 3]
```diff
@@ -99,6 +100,13 @@ def paginate_queryset(self, queryset):
             return None
         return super().paginate_queryset(queryset)
 
+    def get_queryset(self):
+        readable_id = self.request.query_params.get("readable_id", None)
+        if readable_id:
+            return Program.objects.filter(live=True, readable_id=readable_id)
+
+        return Program.objects.filter(live=True)
```
[the remark this answers is not in the document]
Comment (jkachel) [reply courses/views/v1/__init__.py]: I think the filter backend stuff was added after I'd already done this filtering stuff. I redid it to just use the filter backend instead.

File: cms/templatetags/course_list.py (hunk 2 above)
Comment (jkachel) [reply cms/templatetags/course_list.py]: This is fixed in https://github.com/mitodl/mitxonline/pull/1833/commits/af42117971baf01b19def8c36d855c836c60cbef (I noticed it too, then fixed it, probably at the same time you did)

**Comment by jkachel** [comment]
https://github.com/mitodl/mitxonline/pull/1833/commits/f1e062b086fbed76b89789be795153cde8518f3f concludes a set of changes that should resolve comments and also rebases from main (and so should now have the video fixes that got merged in earlier).

**Review by rachellougee — APPROVED** [review]
Tested and noticed two small issues:
1. The height of course card is not fixed, so it looks different from homepage. I think it just needs a `height:100%` for program-course-card?
image
2. The format for start_date is slightly different from design and homepage. It displays format like `Starts Oct 25,2023, 7:17 PM` on program about page vs `Starts October 25, 2023` on homepage.
3. Due to the format difference between frontend and backend, the program start date is slightly off when taking the earliest start date of the courses in the program
image

 Everything else seems work as expected.

**Pull request merged**